docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+---
+id: intro-react
+title: Key React Concepts
+---
+
+React Native runs on [React](https://reactjs.org/), a popular open source library for building user interfaces with JavaScript. To make the most of React Native, it helps to understand React itself. This section can get you started or can serve as a refresher course.
+
+We’re going to cover the core concepts behind React:
+
+- components
+- JSX
+- custom components
+- props
+- state
+
+If you want to dig deeper, we encourage you to check out [React's official documentation](https://reactjs.org/docs/getting-started.html).
+
+## Your first component
+
+The rest of this introduction to React uses cats in its examples: friendly, approachable creatures that need names and a cafe to work in. Here is your very first Cat component:
+
+<div class="toggler">
+  <ul role="tablist" class="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTabs('syntax', 'functional')">
+      Function Component Example
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTabs('syntax', 'classical')">
+      Class Component Example
+    </li>
+  </ul>
+</div>
+
+<block class="functional syntax" />
+
+```SnackPlayer name=Your%20Cat
+import React from 'react';
+import { Text } from 'react-native';
+
+export default function Cat() {
+  return (
+    <Text>Hello, I am your cat!</Text>
+  );
+}
+```
+
+Here is how you do it: To define your `Cat` component, first use JavaScript’s [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) to import React and React Native’s [`Text`](/react-native/docs/next/text) Core Component:
+
+```jsx
+import React from 'react';
+import {Text} from 'react-native';
+```
+
+Your component starts as a function:
+
+```jsx
+function Cat() {}
+```
+
+Whatever a function component returns is rendered as a React element. `Cat` will render a `<Text>` element:
+
+```jsx
+function Cat() {
+  return <Text>Hello, I am your cat!</Text>;
+}
+```
+
+You can export your function component with JavaScript’s [`export`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) for use throughout your app like so:
+
+```jsx
+export default function Cat() {
+  // element(s) go here!
```

Comment:
This snippet is focused on introducing `export default`. However, you're stealing attention from that by replacing `return <Text>Hello, I am your cat!</Text>;` from previous example with `// element(s) go here!`. Let's just keep the function contents as is now, since we already introduced the elements. Then `export default` will be more noticeable.

Maybe even:
```jsx
// We added "export default"
export default function Cat() {
  return <Text>Hello, I am your cat!</Text>;
}
```

Reply:
Sharp eyes! I'd like to know how you highlight lines of code in the React docs, as it would be great to also highlight the `export` line!